### ISSUE TITLE
Potential solution for configurator.py

### ISSUE THREAD

**Issue by stllfe** [karpathy/nanochat#166]
Hey there @karpathy! Thank you for yet another amazing learning and inspiration resource!

I've noticed that your "poorman's configuration system" is used here as well. And honestly if it gets the job done I think it's easily justifiable. But for my ML experiments I've wanted to make something elegant and simple as well and I guess I kinda achieved that, or at least pretty close to something usable now. 

I've made a simple library called [manuscript](https://github.com/stllfe/manuscript) (which itself is based on [tyro](https://github.com/brentyi/tyro)). It's inspired by your on-the-fly code patching/inline exec approach where it reads the `with` block and converts all variables defined within this section as configurable through CLI, intercepts the argparse and injects the values back. So like the following:
```python
import manu
import manu.conf as conf

with manu.script as args:
  exp_name: str = ...  # required
  out_dir: Path = ...  # rich types supported
  device = "cpu"  # with default, type hint is optional
  learning_rate = 6e-4  # inline comments become help text
  beta = 0.9  #! comments with ! are ignored
  max_iters = 6000000
  """this is a docstring for max_iters as well"""
  do_something: conf.Fixed[bool] = True  #! configure args via hints

# at this point `exp_name` is an actual runtime string
# so you can use it as usual
print("Using exp name:", exp_name)

# you can also do something with all the captured values
print("All CLI variables:")
print(args)  # Mapping[str, Any]

# log final config to wandb, mlflow, etc.
# args is basically a mapping of all variables and their final values
wandb.config.update(manu.script.values)  # same as args
```

I intend it to be a hustle free and pythonic configuration system for flat imperative style CLI scripts. 

I don't want to plug this forecefully just wanted to share. Maybe you or someone reading this would find it useful, thanks!

**Comment by andreeabogdanovici** [comment]
over-engineering the config layer for aesthetics doesn’t add function. just adds new headaches when it breaks & you need exact replication. dependency injection = you’re looking through 39567 files for what the duck got overridden. so function beats form. the ‘poor man’s config’ works because it’s dumb & explicit. one can crash, fix, get the same loss curve bit for bit. diff in 2s, reproduce the exact seed & hyperparams, know what ran. it’s dumb, yes, but elegance in how it fails.

**Comment by karpathy** [comment]
I moved all of this to argparse now ty!
